core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -49,74 +46,38 @@ class ReplicaFetcherThread(name: String,
                            replicaMgr: ReplicaManager,
                            metrics: Metrics,
                            time: Time,
-                           quota: ReplicationQuotaManager)
+                           quota: ReplicationQuotaManager,
+                           leaderEndpointBlockingSend: Option[BlockingSend] = None)
   extends AbstractFetcherThread(name = name,
                                 clientId = name,
                                 sourceBroker = sourceBroker,
                                 fetchBackOffMs = brokerConfig.replicaFetchBackoffMs,
-                                isInterruptible = false) {
+                                isInterruptible = false,
+                                includePartitionInitialisation = brokerConfig.interBrokerProtocolVersion >= KAFKA_0_11_0_IV2) {
```

Comment:
Hmm, if the broker is on a version before KAFKA_0_11_0_IV2 and we don't let the partition go through the initialization phase, then the followers won't be doing any truncation based on HW. 

We can probably always set the partition to need initialization. If the broker is on a version before KAFKA_0_11_0_IV2, in fetchEpochsFromLeader(), we don't do the actual leader epoch request, but simply set the response to UNKNOWN_OFFSET. Then the maybeTruncate() logic will just fall back to HW.